src/interface/blockchain.cpp

```diff
@@ -63,7 +62,7 @@ void blockchain::fetch_history4(server_node& node, const message& request,
     const auto script_hash = deserial.read_hash();
     const size_t from_height = deserial.read_4_bytes_little_endian();
 
-    node.chain().fetch_history(script_hash, limit, from_height,
+    node.chain().fetch_history(script_hash, default_limit, from_height,
```

Comment:
The zero limit should be implemented a sentinel for unlimited so that this is not required.

Reply:
Ok.  This is related to https://github.com/libbitcoin/libbitcoin-blockchain/blob/master/src/interface/block_chain.cpp#L1539

Reply:
Since the from_height is limiting somewhat, maybe the limit param isn't needed in particular?  Or is it still a good sanity check on the receiving end for highly active histories?